src/Microsoft.AspNetCore.ResponseCompression/CompressionProviderCollection.cs

```diff
@@ -0,0 +1,51 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Collections.ObjectModel;
+#if NETSTANDARD1_3
+using System.Reflection;
+#endif
+
+namespace Microsoft.AspNetCore.ResponseCompression
+{
+    /// <summary>
+    /// A Collection of ICompressionProvider's that also allows them to be instantiated from DI.
+    /// </summary>
+    public class CompressionProviderCollection : Collection<ICompressionProvider>
```

Comment:
Wasn't part of the design, so I'm not super familiar. Would we want to consider a design like `FilterCollection` in Mvc since it looks pretty similar - https://github.com/aspnet/Mvc/blob/760c8f38678118734399c58c2dac981ea6e47046/src/Microsoft.AspNetCore.Mvc.Core/Filters/FilterCollection.cs?

Essentially,

* `ICompressionProviderMetadata`, 
* `ICompressionProvider : ICompressionProviderMetadata`
* `ICompressionProviderFactory : ICompressionProviderMetadata` 
* and this would be `Collection<ICompressionProviderMetadata>`

Obviously, filters are a much more useful extensibility point and I'm not sure this is.